server/http_test.go

```diff
@@ -732,3 +734,131 @@ func TestHTTPServer_EnvironmentBlocks(t *testing.T) {
 		t.Errorf("Unexpected status code: %d", res.StatusCode)
 	}
 }
+
+func TestHTTPServer_RateLimiterFixed(t *testing.T) {
+	helper := test.New(t)
+	client := newClient()
+
+	shutdown, hook := newCouper("testdata/integration/ratelimit/01_couper.hcl", test.New(t))
+	defer shutdown()
+
+	req, err := http.NewRequest(http.MethodGet, "http://anyserver:8080/fixed", nil)
+	helper.Must(err)
+
+	hook.Reset()
+
+	go client.Do(req)
+	go client.Do(req)
+	go client.Do(req)
+
+	time.Sleep(1200 * time.Millisecond)
+
+	entries := hook.AllEntries()
+	if len(entries) != 6 {
+		t.Fatal("Missing log lines")
+	}
+
+	for i, entry := range entries {
+		if i%2 == 0 {
+			continue
+		}
+
+		if total := entry.Data["timings"].(logging.Fields)["total"].(float64); total <= 0 {
+			t.Fatal("Something is wrong")
+		} else if i < 4 && total > 1000 {
+			t.Errorf("Request time has to be smaller as 1 second")
+		} else if i >= 4 && total < 1000 {
+			t.Errorf("Request time has to be longer as 1 second")
+		}
+	}
+}
+
+func TestHTTPServer_RateLimiterSliding(t *testing.T) {
```

Comment:
Also test the _differences_ between `period_window = "fixed"` and `period_window = "sliding"`.

Reply:
Done